src/anomalib/callbacks/metrics.py

```diff
@@ -85,9 +84,22 @@ def setup(
             )
         else:
             pixel_metric_names = (
-                self.pixel_metric_names if not isinstance(self.pixel_metric_names, str) else [self.pixel_metric_names]
+                self.pixel_metric_names.copy()
+                if not isinstance(self.pixel_metric_names, str)
+                else [self.pixel_metric_names]
             )
 
+        semantic_pixel_metric_names: list[str] | dict[str, dict[str, Any]]
+        if "SPRO" in pixel_metric_names:
+            if isinstance(pixel_metric_names, list):
+                pixel_metric_names.remove("SPRO")
+                semantic_pixel_metric_names = ["SPRO"]
+            elif isinstance(pixel_metric_names, dict):
+                spro_metric = pixel_metric_names.pop("SPRO")
+                semantic_pixel_metric_names = {"SPRO": spro_metric}
+            else:
+                logger.warning("Unexpected type for pixel_metric_names: %s", type(pixel_metric_names))
```

Comment:
Would it be an idea to add a comment here explaining why we are doing this. Would be good to have it for future reference, and for external readers.

Reply:
done